src/TestableHttpClient/Utils/RouteParser.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System.IO;
+
+namespace TestableHttpClient.Utils;
+
+internal static class RouteParser
+{
+    public static RouteDefinition Parse(string pattern)
+    {
+        if (string.IsNullOrEmpty(pattern))
+        {
+            throw new RouteParserException();
+        }
+
+        ReadOnlySpan<char> patternSpan = pattern.AsSpan();
+        RouteDefinition routeDefinition = RouteDefinition.Any;
+        int currentPosition = 0;
+        char currentChar = patternSpan[currentPosition];
+
+        if (currentChar == '*' && patternSpan.Length == 1)
```

Comment:
Get rid of `currentChar` by checking length or by matching the span with `['*']`